torch/csrc/distributed/rpc/python_rpc_handler.cpp

```diff
@@ -56,33 +72,43 @@ std::vector<char> PythonRpcHandler::generatePythonUDFResult(
 py::object PythonRpcHandler::loadPythonUDFResult(
     const std::vector<char>& pickledPayload,
     const std::vector<torch::Tensor>& tensorTable) {
+  GilWaitTimeGuard g;
   pybind11::gil_scoped_acquire ag;
+  g.markAcquired();
   auto pargs = py::bytes(pickledPayload.data(), pickledPayload.size());
   return pyLoadReturnValue_(pargs, tensorTable);
 }
 
 py::object PythonRpcHandler::runPythonUDF(
     const SerializedPyObj& serializedObj) {
+  GilWaitTimeGuard g;
   pybind11::gil_scoped_acquire ag;
+  g.markAcquired();
```

Comment:
Can we make this optional such that it can be enabled/disabled via flag (default can be enabled for now)? There is quite a bit of overhead in `markAcquired`. Also, this block of code could be replaced with a MACRO (which performs appropriate guards based on whether or not this feature is enabled) instead of repeating these three lines everywhere.

Reply:
@pritamdamania87 Did you mean a compile-time flag or a runtime variable that can be toggled?

Reply:
@pritamdamania87 Went ahead and made it configurable at runtime.